File: challenge-374/abigail/python/ch-2.py
#!/opt/homebrew/bin/python3

import sys
import re

for line in sys . stdin:
    #
    # Extract all sequences of the same digit from the string,
    # map them to integers, then find and print the maximum.
    #
    print (max (map (lambda x: int (x [0]),
                     re . findall (r'(([0-9])\2*)', line))))

File: challenge-374/abigail/node/ch-2.js
#!/opt/homebrew/bin/node
 
  require ('readline')
. createInterface ({input: process . stdin})
. on              ('line', line => {
      //
      // Find all sequences of the same digit, numify them,
      // then use reduce to find the maximum
      //
      console . log ([... line . matchAll (/([1-9])\1*/g)] .
                      map    (x => +x [0])                 .
                      reduce ((seq, max) => Math . max (seq, max), 0))
  })
